### ISSUE TITLE
/usr/lib/dyld: cannot execute binary file error on macOS + nix-shell

### ISSUE THREAD

**Issue by dimitriho** [thesamet/sbt-protoc#253]
**How to reproduce**

- Update the versions in `examples/scalapb/project/plugins.sbt`
```
    addSbtPlugin("com.thesamet" % "sbt-protoc" % "1.0.4")
    
    libraryDependencies += "com.thesamet.scalapb" %% "compilerplugin" % "0.11.4"
```

- run `nix-shell` on the git root directory, then `cd examples/scalapb && sbt compile`

=> protocGenerate task fails with `/usr/lib/dyld: cannot execute binary file`

The latest working version for sbt-protoc was 0.99.34. I get this error with 1.0.0-RC1 and later.

**Comment by dimitriho** [comment]
So I just read the changelog in commit 20b2a2a, it appears that running `unset NIX_CC` before `sbt compile` fixes this error.

**Comment by thesamet** [comment]
Thanks for reporting it. Sounds like our Nix detection has an undesirable side-effect. I guess the logic needs to be `NIX_CC` defined and OS is Linux.

**Comment by thesamet** [comment]
I released protoc-bridge v0.9.3 with your PR. The next step would be to bump up the version here and leverage the new public method. Will you be able to send a PR?

**Comment by dimitriho** [comment]
Sure I'll update this PR, thanks for the release
